src/app/modules/payments/__tests__/stripe.controller.spec.ts

```diff
@@ -209,4 +218,141 @@ describe('stripe.controller', () => {
+      await dbHandler.clearCollection(Payment.collection.name)
+      await dbHandler.clearCollection(EncryptPendingSubmission.collection.name)
+    })
+    const mockForm = {
+      _id: MOCK_FORM_ID,
+    } as IPopulatedForm
+    it('should return UNPROCESSABLE_ENTITY when state mismatch', async () => {
+      // Arrange
+      const mockReq = expressHandler.mockRequest({
+        query: { state: 'otherState', code: 'someCode' },
+        others: { signedCookies: { stripeState: 'anotherState' } },
+      })
+      const mockRes = expressHandler.mockResponse()
+      // Act
+      await StripeController._handleConnectOauthCallbackForTest(
+        mockReq,
+        mockRes,
+        jest.fn(),
+      )
+
+      // Assert
+      expect(mockRes.status).toHaveBeenCalledWith(
+        StatusCodes.UNPROCESSABLE_ENTITY,
+      )
+      expect(mockRes.json).toHaveBeenCalledWith({
+        message: 'Invalid state parameter',
+      })
+    })
+
+    it('should redirect back to settings/payment page when code is undefined', async () => {
+      // Arrange
+      const mockReq = expressHandler.mockRequest({
+        query: { state: 'otherState' },
+        others: { signedCookies: { stripeState: 'otherState' } },
+      })
+      const mockRes = expressHandler.mockResponse()
+      // Act
+      await StripeController._handleConnectOauthCallbackForTest(
+        mockReq,
+        mockRes,
+        jest.fn(),
+      )
+
+      // Assert
+      expect(mockRes.redirect).toHaveBeenCalledWith(
+        `${config.app.appUrl}/admin/form/otherState/settings/payments`,
+      )
+    })
+
+    it('should redirect back to settings/payment page when stripe account linking is successful', async () => {
+      // Arrange
+      const mockReq = expressHandler.mockRequest({
+        query: { state: MOCK_FORM_ID + '.otherState', code: 'someCode' },
+        others: {
+          signedCookies: { stripeState: MOCK_FORM_ID + '.otherState' },
+        },
+      })
+      MockFormService.retrieveFullFormById.mockReturnValue(okAsync(mockForm))
+      const mockStripeToken = {
+        stripe_user_id: 'user_id',
+        stripe_publishable_key: 'publishable_key',
+      } as MarkRequired<
+        Stripe.OAuthToken,
+        'stripe_user_id' | 'stripe_publishable_key'
+      >
+      MockEncryptSubmissionService.checkFormIsEncryptMode.mockReturnValue(
+        ok(mockForm as IPopulatedEncryptedForm),
+      )
+      MockStripeService.exchangeCodeForAccessToken.mockReturnValue(
+        okAsync(mockStripeToken),
+      )
+      MockStripeService.linkStripeAccountToForm.mockReturnValue(
+        okAsync('someId'),
+      )
+      const mockRes = expressHandler.mockResponse()
+      // Act
+      await StripeController._handleConnectOauthCallbackForTest(
+        mockReq,
+        mockRes,
+        jest.fn(),
+      )
+
+      // Assert
+      expect(mockRes.redirect).toHaveBeenCalledWith(
+        `${config.app.appUrl}/admin/form/${MOCK_FORM_ID}/settings/payments`,
```

Comment:
shall we abstract this out to the top level of the test?

Reply:
Oh was thinking to abstract out `${config.app.appUrl}/admin/form/${MOCK_FORM_ID}/settings/payments` like

`EXPECTED_REDIRECT_URL = ${config.app.appUrl}/admin/form/${MOCK_FORM_ID}/settings/payments`

Cause it's referred to multiple times, and the string by itself is a little 'magicky'

But small thing no worries HAHAHA since its merged its okay de!